aten/src/ATen/core/DeprecatedTypePropertiesRegistry.h

```diff
@@ -5,40 +5,29 @@
 
 #include <c10/core/Backend.h>
 #include <c10/core/ScalarType.h>
-#include <ATen/core/DeprecatedTypeProperties.h>
 
 namespace at {
 
+struct DeprecatedTypeProperties;
+
 struct CAFFE2_API DeprecatedTypePropertiesDeleter {
-  void operator()(DeprecatedTypeProperties * ptr) {
-      delete ptr;
-  }
+  void operator()(DeprecatedTypeProperties * ptr);
 };
 
 class CAFFE2_API DeprecatedTypePropertiesRegistry {
  public:
   using DeprecatedTypePropertiesUniquePtr =
       std::unique_ptr<DeprecatedTypeProperties, DeprecatedTypePropertiesDeleter>;
```

Comment:
Reminding myself: the reason the deleter is explicitly specified here, is because we only have a forward declaration of DeprecatedTypeProperties at this point (so the default deleter won't work); is that right?

Reply:
That sounds right. But now that you mention it I actually don't need it anymore because I moved definitions to the cpp file.